agimus_controller/agimus_controller/warm_start_shift_previous_solution.py

```diff
@@ -0,0 +1,97 @@
+"""This module defines the WarmStartShiftPreviousSolution class.
+
+The next warm start is generated based only on the previous solution. It
+shifts the previous solution by the amount of the first time step.
+- It supports non-constant time steps.
+- It assumes the actuation model and the dynamics is the same all along the
+  trajectory. To overcome this issue, it would need to take the models from
+  the OCP.
+
+When there is no previous solution, the warm start is calculated using an internal
+WarmStartReference object.
+"""
+
+import numpy as np
+import numpy.typing as npt
+import pinocchio
+import crocoddyl
+
+from agimus_controller.trajectory import TrajectoryPoint
+from agimus_controller.warm_start_base import WarmStartBase
+
+
+class WarmStartShiftPreviousSolution(WarmStartBase):
```

Comment:
I'll open another PR where I fix this class so that it works. Right now, it is not working properly.